nautobot/ipam/models.py

```diff
@@ -1037,6 +1064,19 @@ def clean(self):
         # Force dns_name to lowercase
         self.dns_name = self.dns_name.lower()
 
+    def save(self, *args, **kwargs):
+        if not self.present_in_database:
+            if self._namespace is None:
+                raise ValidationError({"parent": "Namespace could not be determined."})
+            namespace = self._namespace
+        else:
+            namespace = self.parent.namespace
+
+        # Determine the closest parent automatically based on the Namespace.
+        self.parent = Prefix.objects.get_closest_parent(self.host, namespace=namespace)
```

Comment:
Do we need to recalculate this on every save? Or should this only be done on create?

Reply:
At this time it's entirely possible to edit the IP address or change its namespace, so unless we decide to make such properties immutable after create we need this to be recalculated on every save.